ide/src/main/java/com/ibm/wala/ide/classloader/EclipseSourceFileModule.java

```diff
@@ -17,17 +17,13 @@
 /** A module which is a wrapper around a .java file */
 public class EclipseSourceFileModule extends SourceFileModule {
 
-  private final IFile f;
+  protected final IFile f;
 
-  public static EclipseSourceFileModule createEclipseSourceFileModule(IFile f) {
-    if (f == null) {
-      throw new IllegalArgumentException("null f");
-    }
-    return new EclipseSourceFileModule(f);
-  }
-
-  private EclipseSourceFileModule(IFile f) {
-    super(new File(f.getFullPath().toOSString()), f.getName(), null);
+  public EclipseSourceFileModule(IFile f) {
+    super(
```

Comment:
Can you add a comment here indicating that passing `null` to `super` will cause a failure in the `FileModule` constructor?  We should really get NullAway running on all this code to get rid of this kind of stuff :-)

Reply:
Does it cause a failure? We would hit this code:

https://github.com/wala/WALA/blob/3a2ca2e599e1cf72073ed64d5cf5839dea93bf71/core/src/main/java/com/ibm/wala/classLoader/FileModule.java#L30-L32

Which is what we had previously. The only thing that would really be different is that the filename string would be null.

Reply:
Or, is the thought that it was clear in this file that an exception would be thrown in this case, and now with the new code, we would need to go two levels up to see the same thing?

Reply:
My point was that it'd be good to write a comment indicating that we deliberately pass `null` _knowing_ that there is logic to fail elsewhere, so a reader does not get confused and think it is valid to pass `null` to the super constructor